feedly/api_client/session.py

```diff
@@ -125,25 +125,57 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         self.close()
 
     @property
-    def user(self) -> "FeedlyUser":
+    def user(self) -> FeedlyUser:
         return self._user
 
     def do_api_request(
-        self, relative_url: str, method: str = None, data: Dict = None, timeout: int = None, max_tries: int = None
-    ) -> Union[Dict[str, Any], List[Any]]:
+        self,
+        relative_url: str,
+        method: str = None,
+        params: Dict[str, Any] = None,
+        data: Dict = None,
+        timeout: int = None,
+        max_tries: int = None,
+    ) -> Union[Dict[str, Any], List[Any], None]:
         """
-        makes a request to the feedly cloud API (https://developers.feedly.com)
+        makes a request to the feedly cloud API (https://developers.feedly.com), and parse the response
         :param relative_url: the url path and query parts, starting with /v3
+        :param params: the query parameters.
         :param data: the post data to send (as json).
         :param method: the http method to use, will default to get or post based on the presence of post data
         :param timeout: the timeout interval
         :param max_tries: the number of tries to do before failing
-        :param protocol: the protocol to use (http or https)
         :return: the request result as parsed from json.
         :rtype: dict or list, based on the API response
         :raises: requests.exceptions.HTTPError on failure. An appropriate subclass may be raised when appropriate,
          (see the ones defined in this module).
         """
+        resp = self.make_api_request(
+            relative_url=relative_url, method=method, params=params, data=data, timeout=timeout, max_tries=max_tries
+        )
+        return resp.json() if resp.content is not None and len(resp.content) > 0 else None
+
+    def make_api_request(
+        self,
+        relative_url: str,
+        method: str = None,
+        params: Dict[str, Any] = None,
+        data: Dict = None,
+        timeout: int = None,
+        max_tries: int = None,
+    ) -> Response:
```

Comment:
I need to access the response headers to get the continuation in the IoC use case